### PR TITLE
{vis}[foss/2018a] MIGRATE-N 4.2.14

### PR DESCRIPTION
**Pull request by RvDijk**

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
dell.lan - Linux fedora 27, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz, Python 2.7.14
See https://gist.github.com/8e3eb9a9105ba9e698cec281f6e2ec5d for a full test report.

**Comment by verdurin** [comment]
@RvDijk build error:
```
Create the PDF library (libharu 1.x): [linker warnings are OK]
Using system zlib architecture
Compile the main source files and create the MPI-parallel executable
src/data.c:57:21: fatal error: xlocale.h: No such file or directory
 #include <xlocale.h>
                     ^
compilation terminated.
make[1]: *** [Makefile:195: data.o] Error 1
make: *** [Makefile:262: mpis] Error 2
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2569.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/8f3e11ad6de60ad9d8e5a93ea964f159 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2047.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/703027691fe6faf9449413ddc2b0cf1b for a full test report.

**Comment by RvDijk** [comment]
xlocale.h is also in the OS?
``` 
rpm -q --whatprovides /usr/include/xlocale.h
glibc-headers-2.17-196.el7_4.2.x86_64
```

**Comment by verdurin** [comment]
Presumably another CentOS-6 related problem?:
```
rpm -q --whatprovides /usr/include/xlocale.h
glibc-headers-2.12-1.192.el6.x86_64
```

**Comment by verdurin** [comment]
Forget that, I was building on a Fedora 27 machine, which doesn't have `xlocale.h`.

@RvDijk - the point is that you should really specify this as a dependency within the `easyconfig`, rather than relying on the OS package. This isn't a security-sensitive package, so I don't see a good reason for this to be an OS dependency.

**Comment by RvDijk** [comment]
@verdurin how to specify this as a dependency? Maybe a dependency on https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/g/glibc/glibc-2.26-GCCcore-6.4.0.eb ?

**Comment by boegel** [comment]
Is this thing trying to build its own `libharu`? We have easyconfig to provide that.

You could try including `glibc` as a dependency, but I doubt that'll be enough (unless it's only about providing the headers, in which case you may be able to get away with including `glibc` as a build dep).

Usually when you include a newer version of `glibc`, you need to do it for *everything* (so it basically needs to be part of the toolchain).

**Comment by RvDijk** [comment]
@verdurin could you check if libharu dependency gets picked up? If not I'll probably have to write a patch...

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
dell.lan - Linux fedora 27, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz, Python 2.7.14
See https://gist.github.com/51533205f55e640320fb4ca958b97816 for a full test report.

**Comment by verdurin** [comment]
@RvDijk `libharu` is picked up and built. Still seeing the same `xlocale.h` error.

**Comment by boegel** [comment]
Since the `Create the PDF library (libharu 1.x): [linker warnings are OK]` message is still there, it's unclear to me whether the `libharu` we provided is picked up (and whether it's actually needed at all...)

It does seem like this version of `MIGRATE-N` is simply not compatible with older versions of `glibc` though.

It seems like it may be an issue with more recent versions of `glibc` too though, see https://github.com/Microsoft/cpprestsdk/issues/485 and https://github.com/spack/spack/issues/5713...

**Comment by RvDijk** [comment]
@verdurin could you check this one as well? I don't have a Fedora system available for testing.

File: easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+homepage = 'http://libharu.org/'
+description = """libHaru is a free, cross platform, open source library for generating PDF files."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
+
+source_urls = [' https://github.com/libharu/libharu/archive/']
+sources = ['RELEASE_%s.tar.gz' % '_'.join(version.split('.'))]
+
+dependencies = [('libpng', '1.6.32')]
```
Comment (boegel) [inline easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb]: @RvDijk We're going with `libpng` 1.6.34 in easyconfigs that use `2018a` toolchain

File: easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    'migrate-%(version)s_install.patch',
+]
+checksums = [
+    'a4387dda154d5df6d8943b6edba44e9d29595f1a5853a5909f69c4caf7aed7c3',  # migrate-4.2.14.src.tar.gz
+    'b84eea6c8e76af3ef2a71aa566962cf6e49c3ec9c808993d85dc74e86e530c94',  # migrate-4.2.14_install.patch
+]
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('glibc', '2.17'),
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb]: @RvDijk How well did you test the result installation? You typically need to be very careful when including your own `glibc`... Is it an option to make this a build dep?

Comment (RvDijk) [reply easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb]: @boegel I'll look into some testing now, probably build dep is preferred here.

File: easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb (hunk 1 above)
Comment (RvDijk) [reply easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb]: @boegel libharu is not going to be used, do you still want to include this in this PR?

File: easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb (hunk 2 above)
Comment (RvDijk) [reply easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb]: @boegel on this machine including glibc is breaking everything (also as a build dep) ... 
Probably and hopefully the `xlocale.h` inclusion isn't necessary, so I'll try a patch that will exclude it.

File: easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/l/libharu/libharu-2.3.0-foss-2018a.eb]: No, let's remove it.

File: easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    'http://popgen.sc.fsu.edu/currentversions/',
+    'http://popgen.sc.fsu.edu/oldversions/%(version_major)s.x/%(version_major_minor)s',
+    'http://popgen.sc.fsu.edu/oldversions/',
+    'http://popgen.sc.fsu.edu/newversions/',
+]
+
+patches = [
+    'migrate-%(version)s_install.patch',
+    'migrate-%(version)s_xlocale.patch',
+]
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE-N/MIGRATE-N-4.2.14-foss-2018a.eb]: @RvDijk Please include `SHA256` checksums for both `sources` and `patches` (easiest way is to run `eb --inject-checksums`).

**Comment by boegel** [comment]
@verdurin Can you look into a new test report for this?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2627.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4c45da8c5b828c58035cde560dc54e24 for a full test report.

**Pull request merged**